src/controllers/server.js

```diff
@@ -72,13 +72,17 @@ class Server extends EventEmitter {
         this.lastCrash = undefined;
         this.failedQueryCount = 0;
 
-        // @TODO: If container doesn't exist attempt to create a new container and then try again.
-        // If that fails it truly is a fatal error and we should exit.
         this.docker = new Docker(this, (err, status) => {
             if (status) {
                 this.log.info('Daemon detected that the server container is currently running, re-attaching to it now!');
             }
-            return next(err);
+            if (err && err.statusCode === 404) { // no such container
+                this.log.info('Container was not found. Attempting to recreate it.');
+                next(); // Continue normal initialization
+                this.rebuild(() => {
+                    this.log.fatal('Could not recreate container.');
```

Comment:
This logic fails, and will trigger a fatal error even if the rebuild was successful. You should be looking for an `err` object here, and throwing that error into the `this.log.fatal()` function if it exists, otherwise do nothing.

As it stands, even a successful rebuild will throw that error to the console.

```
16:27:03.175Z DEBUG wings: New container created successfully, updating config... (server=67c7a4b0-6f62-46c4-af32-6494d7871ccf)
16:27:03.189Z DEBUG wings: Loaded configuration and initalized server. (server=67c7a4b0-6f62-46c4-af32-6494d7871ccf)
16:27:03.190Z FATAL wings: Could not recreate container. (server=67c7a4b0-6f62-46c4-af32-6494d7871ccf)
```